llvm/lib/Linker/IRMover.cpp

```diff
@@ -595,11 +595,15 @@ Value *IRLinker::materialize(Value *V, bool ForIndirectSymbol) {
   if (!SGV)
     return nullptr;
 
+  // If SGV is from dest, it is already materialized when dest was loaded.
+  if (SGV->getParent() == &DstM)
+    return nullptr;
+
   // When linking a global from other modules than source & dest, skip
   // materializing it because it would be mapped later when its containing
   // module is linked. Linking it now would potentially pull in many types that
   // may not be mapped properly.
-  if (SGV->getParent() != &DstM && SGV->getParent() != SrcM.get())
+  if (SGV->getParent() != SrcM.get())
```

Comment:
It's been awhile since I looked at this code, but from the guards that existed here and the comment block, it sounds like we previously expected that the symbol might be from the dest module. Can you look at how that worked ok for other symbols coming from dest? You might be able to add an assert and run through the llvm tests to see where that case pops up.

Reply:
I added an assert and there's only one test case hitting it. It turns out to be a similar situation but that case does not involve mapping function pointer in metadata so either mapping the global value or not does not affect the correctness.